tests/std/include/test_header_units_and_modules.hpp

```diff
@@ -0,0 +1,1088 @@
+    assert(v[5] == 60);
+}
+
+void test_mutex() {
+    using namespace std;
+    puts("Testing <mutex>.");
+    // see above, tested with <condition_variable>
+    int n       = 10;
+    auto lambda = [&n] { ++n; };
+    once_flag flag;
+    assert(n == 10);
+    call_once(flag, lambda);
+    assert(n == 11);
+    call_once(flag, lambda);
+    assert(n == 11);
+}
+
+void test_new() {
+    using namespace std;
+    puts("Testing <new>.");
+    bool caught_bad_alloc = false;
+
+    try {
+        throw bad_array_new_length{};
+    } catch (const bad_alloc&) {
+        caught_bad_alloc = true;
+    } catch (...) {
+        assert(false);
+    }
+
+    assert(caught_bad_alloc);
+
+    const int* const ptr = new (nothrow) int{1729};
+    assert(ptr);
+    assert(*ptr == 1729);
+    delete ptr;
+
+    static_assert(hardware_constructive_interference_size == 64); // implementation-defined
+    static_assert(hardware_destructive_interference_size == 64); // implementation-defined
+}
+
+void test_numbers() {
+    using namespace std;
+    puts("Testing <numbers>.");
+    static_assert(3.14 < numbers::pi && numbers::pi < 3.15);
+    static_assert(2.71828f < numbers::e_v<float> && numbers::e_v<float> < 2.71829f);
+}
+
+void test_numeric() {
+    using namespace std;
+    puts("Testing <numeric>.");
+    constexpr int arr[]{1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
+    assert(accumulate(begin(arr), end(arr), 0) == 55);
+    static_assert(accumulate(begin(arr), end(arr), 0) == 55);
+}
+
+void test_optional() {
+    using namespace std;
+    puts("Testing <optional>.");
+    constexpr optional<int> opt{in_place, 1729};
+    assert(opt.has_value());
+    assert(opt.value() == 1729);
+    static_assert(opt.has_value());
+    static_assert(opt.value() == 1729);
+
+    constexpr optional<int> empty{nullopt};
+    assert(!empty.has_value());
+    assert(empty.value_or(-1) == -1);
+    static_assert(!empty.has_value());
+    static_assert(empty.value_or(-1) == -1);
+}
+
+void test_ostream() {
+    using namespace std;
+    puts("Testing <ostream>.");
+    const ostream os{nullptr};
+    assert(os.rdbuf() == nullptr);
+}
+
+void test_queue() {
+    using namespace std;
+    puts("Testing <queue>.");
+    queue<int> q;
+    q.push(10);
+    q.push(20);
+    q.push(30);
+    assert(q.size() == 3);
+    assert(q.front() == 10);
+    q.pop();
+    assert(q.front() == 20);
+    q.pop();
+    assert(q.front() == 30);
+    q.pop();
+    assert(q.empty());
+
+    priority_queue<int> pq;
+    pq.push(50);
+    pq.push(10);
+    pq.push(1729);
+    pq.push(200);
+    assert(pq.size() == 4);
+    assert(pq.top() == 1729);
+    pq.pop();
+    assert(pq.top() == 200);
+    pq.pop();
+    assert(pq.top() == 50);
+    pq.pop();
+    assert(pq.top() == 10);
+    pq.pop();
+    assert(pq.empty());
+}
+
+void test_random() {
+    using namespace std;
+    puts("Testing <random>.");
+    minstd_rand0 lcg;
+
+    for (int i = 0; i < 9999; ++i) {
+        (void) lcg();
+    }

```

Comment:
I forgot that existed! It would indeed be simpler to use, and I don't specifically need to test the function call operator here. I'll make that change.